pkg/archive/archive.go

```diff
@@ -1501,3 +1502,19 @@ func (archive *TempArchive) Read(data []byte) (int, error) {
 	}
 	return n, err
 }
+
+// UncompressedDigest returns the digest of the uncompressed tar stream.
+func UncompressedDigest(alg digest.Algorithm, compressedTar io.Reader) (digest.Digest, error) {
```

Comment:
It looks like this is only used in a test, but introduces a new exported function in `pkg/archive` (used in other projects), as well as a new dependency for the package (`github.com/opencontainers/go-digest`).


It feels like "digest" feels not _directly_ relate to "archives" (only in context of handling an archive for an image(-layer). Perhaps we should look at a different package to put this in (could even be in the `internal/testutils` for now if we only use it for tests, or perhaps otherwise in the `image/` package (if we need to have it available elsewhere)?

Reply:
Moved to `internal/testutils`.